docs/search-v2.md

```diff
@@ -143,3 +143,206 @@ sleep(randint(1, 10))
 ```
 
 to `kitsune.search.v2.views.simple_search`.
+
+### Synonyms
+
+The `elasticsearch/dictionaries/synonyms` path contains a text file for each of our search-enabled locales,
+where synonyms are in the
+[Solr format](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/analysis-synonym-graph-tokenfilter.html#_solr_synonyms_2).
+
+`expand` defaults to `True`,
+so synonyms with no explicit mapping resolve to all elements in the list.
+That is to say:
+
+```
+start, open, run
+```
+
+is equivalent to:
+
+```
+start, open, run => start, open, run
+```
+
+It's also worth noting that these synonyms are applied at *query* time,
+not index time.
+
+That is to say,
+if a document contained the phrase:
+
+> Firefox won't play music.
+
+and we had a synonym set up as:
+
+```
+music => music, audio
+```
+
+Then the search query:
+
+> firefox won't play audio
+
+would **not** match that document.
+
+#### Hyponyms and hypernyms (subtypes and supertypes)
+
+The synonym files can also be used to define relations between
+[hyponyms and hypernyms (subtypes and supertypes)](https://en.wikipedia.org/wiki/Hyponymy_and_hypernymy).
+
+For example,
+a user searching for or posting about a problem with Facebook could use the phrase "Facebook isn't working",
+or "social media isn't working".
+Another user searching for or posting about a problem with Twitter could use the phrase "Twitter isn't working",
+or "social media isn't working".
+
+A simple synonym definition like:
+
+```
+social, facebook, face book, twitter
+```
+
+isn't sufficient here,
+as a user querying about a problem with Facebook clearly doesn't have one with Twitter.
+
+Similarly a rule like:
+
+```
+social => social, facebook, face book, twitter
+```
+
+only captures the case where a user has posted about Facebook not working and searched for social media not working,
+not the reverse.
+
+So in this case a set of synonyms should be defined,
+like so:
+
+```
+social, facebook, face book
+social, twitter
+```
+
+With the hypernyms (supertypes) defined across all lines,
+and the hyponyms (subtypes) defined on one line.
+
+This way,
+a search for "social" would also become one for "facebook", "face book" and "twitter".
+Whereas a search for "twitter" would also become one for "social",
+but *not* "facebook" or "face book".
+
+
+#### Interaction with the rest of the analysis chain
+
+All the analyzers which we apply to a document are also applied to the synonyms,
+such as tokenizers, stemmers and stop word filters.
```

Comment:
we should rephrase this, mentioning that the token filters that are defined before the synonyms one will be applied.